arm/Microsoft.Synapse/workspaces/deploy.bicep

```diff
@@ -0,0 +1,264 @@
+  '${encryptionUserAssignedIdentity}': {}
+} : {})
+
+var identityType = !empty(userAssignedIdentitiesUnion) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned'
+
+var identity = {
+  type: identityType
+  userAssignedIdentities: !empty(userAssignedIdentitiesUnion) ? userAssignedIdentitiesUnion : null
+}
+
+var keyVaultUrl = 'https://${encryptionKeyVaultName}${environment().suffixes.keyvaultDns}/keys/${encryptionKeyName}'
+
+var diagnosticsLogs = [for log in logsToEnable: {
+  category: log
+  enabled: true
+  retentionPolicy: {
+    enabled: true
+    days: diagnosticLogsRetentionInDays
+  }
+}]
+
+resource workspace 'Microsoft.Synapse/workspaces@2021-06-01' = {
+  name: name
+  location: location
+  identity: identity
+  tags: tags
+  properties: {
+    azureADOnlyAuthentication: azureADOnlyAuthentication ? azureADOnlyAuthentication : null
+    cspWorkspaceAdminProperties: !empty(initialWorkspaceAdminObjectID) ? {
+      initialWorkspaceAdminObjectId: initialWorkspaceAdminObjectID
+    } : null
+    defaultDataLakeStorage: {
+      accountUrl: 'https://${defaultDataLakeStorageAccountName}.dfs.${environment().suffixes.storage}'
+      filesystem: defaultDataLakeStorageFilesystem
+      createManagedPrivateEndpoint: managedVirtualNetwork ? defaultDataLakeStorageCreateManagedPrivateEndpoint : null
+    }
+    encryption: encryption ? {
+      cmk: {
+        kekIdentity: {
+          userAssignedIdentity: !empty(encryptionUserAssignedIdentity) ? encryptionUserAssignedIdentity : null
+          useSystemAssignedIdentity: encryptionUseSystemAssignedIdentity ? true : false
+        }
+        key: {
+          keyVaultUrl: keyVaultUrl
+          name: encryptionKeyName
+        }
+      }
+    } : null
+    managedResourceGroupName: !empty(managedResourceGroupName) ? managedResourceGroupName : null
+    managedVirtualNetwork: managedVirtualNetwork ? 'default' : null
+    managedVirtualNetworkSettings: managedVirtualNetwork ? {
+      allowedAadTenantIdsForLinking: allowedAadTenantIdsForLinking
+      linkedAccessCheckOnTargetResource: linkedAccessCheckOnTargetResource
+      preventDataExfiltration: preventDataExfiltration
+    } : null
+    publicNetworkAccess: managedVirtualNetwork ? publicNetworkAccess : null
+    purviewConfiguration: !empty(purviewResourceID) ? {
+      purviewResourceId: purviewResourceID
+    } : null
+    sqlAdministratorLogin: sqlAdministratorLogin
+    sqlAdministratorLoginPassword: !empty(sqlAdministratorLoginPassword) ? sqlAdministratorLoginPassword : null
+  }
+}
+
+// Workspace encryption with customer managed keys
+module workspace_cmk '.bicep/nested_cmk.bicep' = if (encryptionActivateWorkspace) {
+  name: '${deployment().name}-cmk'
+  params: {
+    encryptionKeyName: encryptionKeyName
+    encryptionKeyVaultName: encryptionKeyVaultName
+    encryptionKeyVaultResourceGroupName: encryptionKeyVaultResourceGroupName
+    workspaceName: workspace.name
+    workspacePrincipalId: workspace.identity.principalId
```

Comment:
Missing the `usesRbacAuthorization` passthru, doesn't it?

Reply:
Think its sorted with logic in the module when it gets it from the keyvault.